src/audio/module_adapter/module/cadence.c

```diff
@@ -95,6 +113,46 @@ static struct cadence_api cadence_api_table[] = {
 #endif
 };
 
+/* cadence_param_id_fixup - function to support multiple param id spaces.
+ * @mod, current processing module
+ * @param_id, param id received from Host.
+ *
+ * param_id are defined by Cadence per codec. Because param_id might
+ * overlap we cannot use them in the same topology.
+ * in order to fixup this overlap, we introduce a new param id space.
+ *
+ * 0x0000.0000 - 0x7FFF.FFFF, these are original Cadence param ids which
+ * are left 'unfixed'
+ * 0x8000.0000 - 0xFFFF.FFFF, remapped param id space
+ * see `SOF_CADENCE_<codec>_PARAM_ID_START` macros above.
+ */
+static int cadence_param_id_fixup(struct processing_module *mod, uint32_t param_id)
+{
+	uint32_t start;
+	struct cadence_codec_data *cd = module_get_private_data(mod);
+
+	/* param_id already in codec Cadence param space, do nothing */
+	if (param_id < SOF_CADENCE_WRAPPER_PARAM_ID_START)
+		return param_id;
+
+	/* param id uses remapped value, fixup it up to original Cadence value */
+	switch (cd->api_id) {
+	case  CADENCE_CODEC_MP3_DEC_ID:
+		start = SOF_CADENCE_MP3_PARAM_ID_START;
+		break;
+	case  CADENCE_CODEC_AAC_DEC_ID:
+		start = SOF_CADENCE_AAC_PARAM_ID_START;
+		break;
+	default:
```

Comment:
do we really want to do this with code, or should there be a table for remapping? This code would need to be modified for every codec addition, that's not so good.

Reply:
@plbossart i was thinking about a table for remapping. Will try to implement it in next iteration.